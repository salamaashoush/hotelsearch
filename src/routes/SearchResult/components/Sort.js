import React from "react";
import withStyles from "material-ui/styles/withStyles";
import PropTypes from "prop-types";
import Button from "material-ui/Button";

const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  container: {
    display: "flex"
  }
});

const Sort = ({ classes, onChange, sortBy }) => (
  <div className={classes.container}>
    <Button
      data-test="name"
      color={sortBy === "name" ? "primary" : "default"}
      className={classes.button}
      onClick={e => onChange("name")}
    >
      Sort By Name
    </Button>
    <Button
      data-test="price"
      color={sortBy === "price" ? "primary" : "default"}
      className={classes.button}
      onClick={e => onChange("price")}
    >
      Sort By Price
    </Button>
  </div>
);

Sort.propTypes = {
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  sortBy: PropTypes.string.isRequired
};

export default withStyles(styles)(Sort);
