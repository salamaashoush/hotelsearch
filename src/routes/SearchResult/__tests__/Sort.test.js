import React from "react";
import { createMount } from "material-ui/test-utils";
import Sort from "../components/Sort";

const setup = () => {
  const props = {
    classes: {},
    onChange: jest.fn(),
    sortBy: "name"
  };
  let mount = createMount();
  const wrapper = mount(<Sort {...props} />);

  return {
    props,
    wrapper
  };
};
const wrapper = setup().wrapper;
const props = setup().props;
describe(">>> SaerchResult > Sort", () => {
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should changed to price when i changed the prop", () => {
    wrapper.setProps({
      sortBy: "price"
    });
    expect(wrapper).toMatchSnapshot();
  });
});
