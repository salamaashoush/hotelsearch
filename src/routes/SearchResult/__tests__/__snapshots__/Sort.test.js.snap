// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> SaerchResult > Sort should changed to price when i changed the prop 1`] = `
<WithStyles(Sort)
  classes={Object {}}
  onChange={[Function]}
  sortBy="price"
>
  <Sort
    classes={
      Object {
        "button": "Sort-button-1",
        "container": "Sort-container-2",
      }
    }
    onChange={[Function]}
    sortBy="price"
  >
    <div
      className="Sort-container-2"
    >
      <WithStyles(Button)
        className="Sort-button-1"
        color="default"
        data-test="name"
        onClick={[Function]}
      >
        <Button
          className="Sort-button-1"
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-9",
              "dense": "MuiButton-dense-4",
              "disabled": "MuiButton-disabled-15",
              "fab": "MuiButton-fab-16",
              "flatAccent": "MuiButton-flatAccent-7",
              "flatContrast": "MuiButton-flatContrast-8",
              "flatPrimary": "MuiButton-flatPrimary-6",
              "keyboardFocused": "MuiButton-keyboardFocused-11",
              "label": "MuiButton-label-5",
              "mini": "MuiButton-mini-17",
              "raised": "MuiButton-raised-10",
              "raisedAccent": "MuiButton-raisedAccent-13",
              "raisedContrast": "MuiButton-raisedContrast-14",
              "raisedPrimary": "MuiButton-raisedPrimary-12",
              "root": "MuiButton-root-3",
            }
          }
          color="default"
          data-test="name"
          dense={false}
          disableFocusRipple={false}
          disableRipple={false}
          disabled={false}
          fab={false}
          mini={false}
          onClick={[Function]}
          raised={false}
          type="button"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 Sort-button-1"
            data-test="name"
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            keyboardFocusedClassName="MuiButton-keyboardFocused-11"
            onClick={[Function]}
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 Sort-button-1"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-19",
                  "root": "MuiButtonBase-root-18",
                }
              }
              data-test="name"
              disableRipple={false}
              disabled={false}
              focusRipple={true}
              keyboardFocusedClassName="MuiButton-keyboardFocused-11"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <button
                className="MuiButtonBase-root-18 MuiButton-root-3 Sort-button-1"
                data-test="name"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label-5"
                >
                  Sort By Name
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "ripple": "MuiTouchRipple-ripple-24",
                        "rippleFast": "MuiTouchRipple-rippleFast-26",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-25",
                        "root": "MuiTouchRipple-root-20",
                        "wrapper": "MuiTouchRipple-wrapper-21",
                        "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-22",
                        "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-23",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-20"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-20"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
      <WithStyles(Button)
        className="Sort-button-1"
        color="primary"
        data-test="price"
        onClick={[Function]}
      >
        <Button
          className="Sort-button-1"
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-9",
              "dense": "MuiButton-dense-4",
              "disabled": "MuiButton-disabled-15",
              "fab": "MuiButton-fab-16",
              "flatAccent": "MuiButton-flatAccent-7",
              "flatContrast": "MuiButton-flatContrast-8",
              "flatPrimary": "MuiButton-flatPrimary-6",
              "keyboardFocused": "MuiButton-keyboardFocused-11",
              "label": "MuiButton-label-5",
              "mini": "MuiButton-mini-17",
              "raised": "MuiButton-raised-10",
              "raisedAccent": "MuiButton-raisedAccent-13",
              "raisedContrast": "MuiButton-raisedContrast-14",
              "raisedPrimary": "MuiButton-raisedPrimary-12",
              "root": "MuiButton-root-3",
            }
          }
          color="primary"
          data-test="price"
          dense={false}
          disableFocusRipple={false}
          disableRipple={false}
          disabled={false}
          fab={false}
          mini={false}
          onClick={[Function]}
          raised={false}
          type="button"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
            data-test="price"
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            keyboardFocusedClassName="MuiButton-keyboardFocused-11"
            onClick={[Function]}
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-19",
                  "root": "MuiButtonBase-root-18",
                }
              }
              data-test="price"
              disableRipple={false}
              disabled={false}
              focusRipple={true}
              keyboardFocusedClassName="MuiButton-keyboardFocused-11"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <button
                className="MuiButtonBase-root-18 MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
                data-test="price"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label-5"
                >
                  Sort By Price
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "ripple": "MuiTouchRipple-ripple-24",
                        "rippleFast": "MuiTouchRipple-rippleFast-26",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-25",
                        "root": "MuiTouchRipple-root-20",
                        "wrapper": "MuiTouchRipple-wrapper-21",
                        "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-22",
                        "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-23",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-20"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-20"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
    </div>
  </Sort>
</WithStyles(Sort)>
`;

exports[`>>> SaerchResult > Sort should render 1`] = `
<WithStyles(Sort)
  classes={Object {}}
  onChange={[Function]}
  sortBy="name"
>
  <Sort
    classes={
      Object {
        "button": "Sort-button-1",
        "container": "Sort-container-2",
      }
    }
    onChange={[Function]}
    sortBy="name"
  >
    <div
      className="Sort-container-2"
    >
      <WithStyles(Button)
        className="Sort-button-1"
        color="primary"
        data-test="name"
        onClick={[Function]}
      >
        <Button
          className="Sort-button-1"
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-9",
              "dense": "MuiButton-dense-4",
              "disabled": "MuiButton-disabled-15",
              "fab": "MuiButton-fab-16",
              "flatAccent": "MuiButton-flatAccent-7",
              "flatContrast": "MuiButton-flatContrast-8",
              "flatPrimary": "MuiButton-flatPrimary-6",
              "keyboardFocused": "MuiButton-keyboardFocused-11",
              "label": "MuiButton-label-5",
              "mini": "MuiButton-mini-17",
              "raised": "MuiButton-raised-10",
              "raisedAccent": "MuiButton-raisedAccent-13",
              "raisedContrast": "MuiButton-raisedContrast-14",
              "raisedPrimary": "MuiButton-raisedPrimary-12",
              "root": "MuiButton-root-3",
            }
          }
          color="primary"
          data-test="name"
          dense={false}
          disableFocusRipple={false}
          disableRipple={false}
          disabled={false}
          fab={false}
          mini={false}
          onClick={[Function]}
          raised={false}
          type="button"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
            data-test="name"
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            keyboardFocusedClassName="MuiButton-keyboardFocused-11"
            onClick={[Function]}
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-19",
                  "root": "MuiButtonBase-root-18",
                }
              }
              data-test="name"
              disableRipple={false}
              disabled={false}
              focusRipple={true}
              keyboardFocusedClassName="MuiButton-keyboardFocused-11"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <button
                className="MuiButtonBase-root-18 MuiButton-root-3 MuiButton-flatPrimary-6 Sort-button-1"
                data-test="name"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label-5"
                >
                  Sort By Name
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "ripple": "MuiTouchRipple-ripple-24",
                        "rippleFast": "MuiTouchRipple-rippleFast-26",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-25",
                        "root": "MuiTouchRipple-root-20",
                        "wrapper": "MuiTouchRipple-wrapper-21",
                        "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-22",
                        "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-23",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-20"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-20"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
      <WithStyles(Button)
        className="Sort-button-1"
        color="default"
        data-test="price"
        onClick={[Function]}
      >
        <Button
          className="Sort-button-1"
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-9",
              "dense": "MuiButton-dense-4",
              "disabled": "MuiButton-disabled-15",
              "fab": "MuiButton-fab-16",
              "flatAccent": "MuiButton-flatAccent-7",
              "flatContrast": "MuiButton-flatContrast-8",
              "flatPrimary": "MuiButton-flatPrimary-6",
              "keyboardFocused": "MuiButton-keyboardFocused-11",
              "label": "MuiButton-label-5",
              "mini": "MuiButton-mini-17",
              "raised": "MuiButton-raised-10",
              "raisedAccent": "MuiButton-raisedAccent-13",
              "raisedContrast": "MuiButton-raisedContrast-14",
              "raisedPrimary": "MuiButton-raisedPrimary-12",
              "root": "MuiButton-root-3",
            }
          }
          color="default"
          data-test="price"
          dense={false}
          disableFocusRipple={false}
          disableRipple={false}
          disabled={false}
          fab={false}
          mini={false}
          onClick={[Function]}
          raised={false}
          type="button"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 Sort-button-1"
            data-test="price"
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            keyboardFocusedClassName="MuiButton-keyboardFocused-11"
            onClick={[Function]}
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 Sort-button-1"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-19",
                  "root": "MuiButtonBase-root-18",
                }
              }
              data-test="price"
              disableRipple={false}
              disabled={false}
              focusRipple={true}
              keyboardFocusedClassName="MuiButton-keyboardFocused-11"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <button
                className="MuiButtonBase-root-18 MuiButton-root-3 Sort-button-1"
                data-test="price"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label-5"
                >
                  Sort By Price
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "ripple": "MuiTouchRipple-ripple-24",
                        "rippleFast": "MuiTouchRipple-rippleFast-26",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-25",
                        "root": "MuiTouchRipple-root-20",
                        "wrapper": "MuiTouchRipple-wrapper-21",
                        "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-22",
                        "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-23",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-20"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-20"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
    </div>
  </Sort>
</WithStyles(Sort)>
`;
